tools/config/__init__.py

```diff
@@ -653,13 +691,13 @@ def regions(self):
         if  ((self.target.bootloader_img or self.target.restrict_size) and
              (self.target.mbed_app_start or self.target.mbed_app_size)):
             raise ConfigException(
-                "target.bootloader_img and target.restirct_size are "
+                "target.bootloader_img and target.restrict_size are "
                 "incompatible with target.mbed_app_start and "
                 "target.mbed_app_size")
         if self.target.bootloader_img or self.target.restrict_size:
-            return self._generate_bootloader_build(*self.rom)
+            return self._generate_bootloader_build(**self.get_all_active_memories(*self.get_region_list()))
```

Comment:
Given that we only pass in the dict, and we don't have any other arguments, let's drop all of the `*` here.